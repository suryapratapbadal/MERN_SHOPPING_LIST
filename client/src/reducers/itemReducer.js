
const initialState = {
    items: [],
    recipes: [],
    loading: false,
    user: false,
};

const itemReducer = (state = initialState, action) => {
    const { type, payload } = action;
    switch (type) {
        case 'GET_ITEMS':
            return {
                ...state,
                items: payload,
            };

        case 'ADD_ITEM':
            return {
                ...state,
                items: [...state.items, payload],
                loading: false
            };

        case 'DELETE_ITEM':
            return {
                ...state,
                items: state.items.filter(item => item._id !== payload),
                loading: false
            };

        case 'UPDATE_ITEM':
            return {
                ...state,
                items:[...state.items.filter(item => item._id !== payload._id), payload],
                loading: false
            };

        case 'VERIFY_USER':
            return {
                ...state,
                user: true
            };

        case 'LOG_OUT_USER':
            return {
                ...state,
                user: false
            };

        case 'GET_RECIPES':
            return {
                ...state,
                recipes: payload,
            };

        case 'ADD_RECIPE':
            return {
                ...state,
                recipes: state.recipes ? [...state.recipes, payload] : [],
                loading: false
            };

        case 'DELETE_RECIPE':
            return {
                ...state,
                recipes: state.recipes.filter(recipes => recipes._id !== payload),
                loading: false
            };

        case 'LOADING':
            return {
                ...state,
                loading: true,
            };

        default:
            return { ...state };
    }
};

export default itemReducer;