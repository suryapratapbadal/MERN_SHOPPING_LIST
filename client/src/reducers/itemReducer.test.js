import itemReducer from './itemReducer';


describe('select_reducer', () => {
    describe('INITIAL_STATE', () => {
        test('is correct', () => {
          const action = { type: 'dummy_action' };
          const initialState = { items: [], loading: false};
      
          expect(itemReducer(undefined, action)).toEqual(initialState);
          expect(itemReducer(undefined, action)).toMatchSnapshot();
        });
      });

      describe('GET_ITEMS', () => {
        test('returns the correct state', () => {
          const action = { type: 'GET_ITEMS', payload: {items:[{id: 1, name: 'Milk'}]} };
      
          expect(itemReducer(undefined, action)).toMatchSnapshot();
        });
      });

      describe('ADD_ITEM', () => {
        test('returns the correct state', () => {
          const action = { type: 'ADD_ITEM', payload: {items:[{id: 1, name: 'Milk'}]} };
      
          expect(itemReducer(undefined, action)).toMatchSnapshot();
        });
      });

      describe('DELETE_ITEM', () => {
        test('returns the correct state', () => {
          const action = { type: 'DELETE_ITEM', payload: 1};
      
          expect(itemReducer(undefined, action)).toMatchSnapshot();
        });
      });
  });

