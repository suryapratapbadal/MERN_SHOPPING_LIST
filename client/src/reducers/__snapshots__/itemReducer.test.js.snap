// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`select_reducer ADD_ITEM returns the correct state 1`] = `
Object {
  "items": Array [
    Object {
      "items": Array [
        Object {
          "id": 1,
          "name": "Milk",
        },
      ],
    },
  ],
  "loading": false,
}
`;

exports[`select_reducer DELETE_ITEM returns the correct state 1`] = `
Object {
  "items": Array [],
  "loading": false,
}
`;

exports[`select_reducer GET_ITEMS returns the correct state 1`] = `
Object {
  "items": Object {
    "items": Array [
      Object {
        "id": 1,
        "name": "Milk",
      },
    ],
  },
  "loading": false,
}
`;

exports[`select_reducer INITIAL_STATE is correct 1`] = `
Object {
  "items": Array [],
  "loading": false,
}
`;
